### ISSUE TITLE
Allow specifying library search path in `.linkerSetting`

### ISSUE THREAD

**Issue by MaxDesiatov** [apple/swift-package-manager#6710]
### Description

While we support `headerSearchPath` in `cSetting` and `cxxSetting` of target description in package manifests, which I assume corresponds to the common `-I` preprocessor option, it would be great to have a corresponding `librariesSearchPath` in `LinkerSetting` type corresponding to the `-L` linker option.

**Comment by MaxDesiatov** [comment]
@neonichu do you know if this was considered before by any chance?

**Comment by MaxDesiatov** [comment]
Ah, I see `headerSearchPath` doesn't allow absolute paths, which makes potential `librariesSearchPath` much less useful.

**Comment by neonichu** [comment]
Right, it is intentional to not allow absolute paths here, otherwise we breaking the entire build model of packages.

**Comment by compnerd** [comment]
Note: absolute paths are not portable.  We need a way to have a platform specific path representation in the manifest.  The system paths are currently a problem for Windows as there is a forest of roots rather than a single tree.

**Comment by MaxDesiatov** [comment]
Please follow [the "Requiring System Libraries" section in `Usage.md`](https://github.com/apple/swift-package-manager/blob/main/Documentation/Usage.md#requiring-system-libraries) for guidance on how to provide a modulemap file with absolute paths.
